components/team_settings_modal/team_settings_modal.tsx

```diff
@@ -3,22 +3,29 @@
 
 import $ from 'jquery';
 import React from 'react';
-import PropTypes from 'prop-types';
 import ReactDOM from 'react-dom';
 import {Modal} from 'react-bootstrap';
 import {FormattedMessage} from 'react-intl';
 
 import * as Utils from 'utils/utils.jsx';
-const SettingsSidebar = React.lazy(() => import('components/settings_sidebar.tsx'));
+const SettingsSidebar = React.lazy(() => import('components/settings_sidebar'));
 
 import TeamSettings from 'components/team_settings';
 
-export default class TeamSettingsModal extends React.PureComponent {
-    static propTypes = {
-        onHide: PropTypes.func,
-    };
+type Props = {
+    onHide: () => any
```

Comment:
Let's follow the pattern how it's being elsewhere. I'd suggest it to be
```suggestion
    onHide: () => void;
```